src/twisted/conch/ssh/transport.py

```diff
@@ -70,6 +71,15 @@ class _MACParams(tuple):
     """
 
 
+warnings.warn(
+    "Legacy SSH ciphers 'CAST5', 'Blowfish' "
+    "were deprecated in Twisted 22.11. "
```

Comment:
I don't know if raising a warning at import time is the best option...

but I guess that if this is imported it means that SSH is also used.

My concern is to avoid raising the warning if you just have Twisted but for example you use it only for HTTPS.

But I think this is fine.

The warning here, should have the advantage that it is only raised once.

--------------

This needs to use a placeholder.

Version 22.11 is wrong.

```suggestion
    "were deprecated in Twisted NEXT. "
```

Also, maybe move it a bit at the top.

I am not sure what to update.

If you use those ciphers, you should not update :)

The message is a bit confusing.

And what is a "legacy" cipher ? 

Maybe just keept is short.

```
The ciphers based on CAST5 and Blowfish were deprecated in Twisted NEXT and will be removed in a future release.
```

Reply:
> I don't know if raising a warning at import time is the best option...

Isn't reporting a deprecation warning about these ciphers _exactly_ the pre-existing behavior - because that's just what python-cryptography does?

I don't see who benefits from having all of these deprecations reported twice.

Just as bad, like the python-cryptography warnings, the Twisted warnings are 100% inactionable.  There is nothing you can do while using Twisted Conch for SSH to avoid these new warnings.  It doesn't matter if your code is affected or not, you will be shown the warnings.  This dilutes the value of warnings and encourages people to ignore them.

It would really be much better if the warnings were emitted when someone actually _used_ something that's deprecated.  This lets them get rid of the deprecation by removing that use.